bpython/curtsiesfrontend/repl.py

```diff
@@ -844,6 +846,13 @@ def on_control_d(self):
             self.current_line = (self.current_line[:self.cursor_offset] +
                                  self.current_line[(self.cursor_offset + 1):])
 
+    def on_control_o(self):
+        next_idx = -self.rl_history.index + 1
+        next = self.rl_history.entries[next_idx]
+        self.on_enter()
+        print("Next is : " + next)
+        self._set_current_line(next)
```

Comment:
OH one more relevant note: `self._set_current_line` will reset `rl_history.index` as well, so be sure to set the index after this, or use `self._set_current_line(next, reset_rl_history=False)`